lib/internal/bootstrap_node.js

```diff
@@ -114,6 +114,10 @@
       }
 
       if (process._eval != null && !process._forceRepl) {
+        if (process._syntax_check_only != null) {
+          console.error('--check and --eval flags are mutually exclusive.');
+          process.exit(1);
```

Comment:
The exit code should be `9`, see https://nodejs.org/dist/latest-v7.x/docs/api/process.html#process_exit_codes.

https://github.com/nodejs/node/pull/12087 is doing something similar for the mutually exclusive `--use-bundled-ca` and `--use-openssl-ca` options.